js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,681 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {
+  compile,
+  compileAsync,
+  compileString,
+  compileStringAsync,
+  render,
+  renderSync,
+  nodePackageImporter,
+  NodePackageImporter,
+  LegacyException,
+  LegacyResult,
+  FileImporter,
+} from 'sass';
+
+import {sandbox, SandboxDirectory} from './sandbox';
+
+const fileImporter = (dir: SandboxDirectory): FileImporter => ({
+  findFileUrl: file => dir.url(file),
+});
+
+function manifestBuilder(overrides?: {[key: string]: string | Object}) {
+  return JSON.stringify({
+    name: 'foo',
+    version: '1.0.0',
+    description: 'Verifying proposed pkg: loading algorithm',
+    scripts: {
+      test: 'echo "Error: no test specified" && exit 1',
+    },
+    author: 'Package Author',
+    license: 'ISC',
+    ...overrides,
+  });
+}
+
+function exportsBuilder(key: string) {
+  return {
+    exports: {
+      '.': {
+        [key]: './src/sass/_styles.scss',
+      },
+      './colors/index.scss': {
+        [key]: './src/sass/colors/index.scss',
+      },
+      './_variables.scss': {
+        [key]: './src/sass/_variables.scss',
+      },
+    },
+  };
+}
+
+describe('Node Package Importer', () => {
+  describe('resolves conditional exports', () => {
+    ['sass', 'style', 'default'].forEach(key => {
+      it(`${key} at root`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder(
+              exportsBuilder(key)
+            ),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+      it(`${key} at root without non-pathed exports`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': manifestBuilder({
+              exports: {
+                [key]: './src/sass/_styles.scss',
+              },
+            }),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+
+      it(`${key} with subpath`, () =>
+        sandbox(dir => {
+          dir.write({
+            'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+            'node_modules/foo/src/sass/_variables.scss': 'a {b: c}',
```

Comment:
Calling this `_variables` when it doesn't contain any variables is a bit confusing. We often use "other" in tests when there are two different stylesheets that might be loaded.